src/codec/encoder/PyEncoder.cpp

```diff
@@ -166,13 +156,23 @@ void PyEncoder::init(prtx::GenerateContext& /*context*/) {
  * preparator. In case the shape generation fails, we collect the initial shape.
  */
 void PyEncoder::encode(prtx::GenerateContext& context, size_t initialShapeIndex) {
+	prtx::EncodePreparator::PreparationFlags enc_prep_flags =
+	        prtx::EncodePreparator::PreparationFlags()
+	                .instancing(false)
+	                .triangulate(false)
+	                .mergeVertices(false)
+	                .cleanupUVs(false)
+	                .cleanupVertexNormals(false)
+	                .mergeByMaterial(true); // if false, generation takes ages... 40 sec
+	                                        // instead of 1.5 sec
+
 	const prtx::InitialShape* is = context.getInitialShape(initialShapeIndex);
 	auto* cb = getPyCallbacks(getCallbacks());
 	if (cb == nullptr)
 		throw prtx::StatusException(prt::STATUS_ILLEGAL_CALLBACK_OBJECT);
 
 	if (getOptions()->getBool(EO_TRIANGULATE))
-		ENC_PREP_FLAGS.triangulate(true);
+		enc_prep_flags.triangulate(true);

```

Comment:
you could avoid the if, set the triangulate prep flag directly when you create enc_prep_flags above and make enc_prep_flags const again
```suggestion
		enc_prep_flags.triangulate(getOptions()->getBool(EO_TRIANGULATE));
```